### DISCUSSION TITLE
Can I set an executable flag on an included binary?

### DISCUSSION THREAD

**Discussion by ennerf** [hydraulic-software/conveyor#96] [Q&A] [answered]
I need to include a downloaded executable (w/o executable bit) and need it to be marked executable when inside the package. Is there a mechanism to do that in the config?

Currently I manually download and modify it before including, but it's a bit cumbersome and doesn't work on Windows.

**Comment by mikehearn** [comment]
Unfortunately I don't think there is. The best you can do is wrap your file into a tarball with the correct permissions. Then it'll work even on Windows, because on Windows permissions are propagated through the files on disk using extended attributes. However, there's no way to override those permissions - they're assumed to be correct at input time.

We could pretty easily add a `{ permissions = rwx-r-xr-x }` type field to input definitions, but you'd need to be able to supply the single file as an input.

**Reply by mikehearn** [reply, marked as answer]
We have a prototype change that lets you specify permission changes on remap rules, so you can do e.g. 

```
app.inputs += {
  from = "./some/file"
  remap = "./some/file [+x]"
}
```

Does that look OK?

**Reply by ennerf** [reply]
I don't have time for testing right now, but I think that will solve my problem. Thanks
